components/brave_rewards/renderer/creator_detection_agent.cc

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_rewards/renderer/creator_detection_agent.h"
+
+#include <array>
+#include <optional>
+#include <utility>
+
+#include "base/containers/fixed_flat_map.h"
+#include "base/feature_list.h"
+#include "base/functional/bind.h"
+#include "brave/components/brave_rewards/common/features.h"
+#include "brave/components/brave_rewards/common/publisher_utils.h"
+#include "brave/components/brave_rewards/resources/grit/creator_detection_generated.h"
+#include "content/public/renderer/render_frame.h"
+#include "gin/function_template.h"
+#include "third_party/blink/public/common/associated_interfaces/associated_interface_provider.h"
+#include "third_party/blink/public/common/web_preferences/web_preferences.h"
+#include "third_party/blink/public/platform/platform.h"
+#include "third_party/blink/public/platform/scheduler/web_agent_group_scheduler.h"
+#include "third_party/blink/public/web/blink.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "third_party/blink/public/web/web_script_source.h"
+
+namespace brave_rewards {
+
+namespace {
+
+constexpr auto kScriptMap = base::MakeFixedFlatMap<std::string_view, int>(
+    {{"github.com", IDR_CREATOR_DETECTION_GITHUB_BUNDLE_JS},
```

Comment:
It seems like a problem if we can't update these outside of browser releases. I think they still need to be delivered as a component, no?